Making the Initial Setup Nicer (#198)

s-rah opened the pull request:
This commit makes some updates to the initial setup UI with the aim of
making it a nicer experience for new users.

![image](https://cloud.githubusercontent.com/assets/106626/8145951/f8befc76-11d2-11e5-98b1-dfbdaa2849fc.png)

The initial setup is the first thing a new user of Ricochet sees when
they start up the software - it should be a place where we a) reassure
them that the software is working b) help them perform intiial setup
which we can't figure out and c) help the user form a connection to the
software so we are better able to present intuitive messages to the
user when we do need their input e.g. if the software discovers
something which needs the users attention like their connection to the
Tor  network is blocked.

Presenting complex topics, like proxies and bridges, to a first time
user can be daunting and so the idea behind this initial setup is to
dress up  the initial dialogue and present the user with a sane
default "Press Connect".

There are still issues with this initial setup - in particular when there
is additional configuration needed - we need to refine ways to work with
the user and, where we can, offer suggestions in a safe and secure
manner.

Image copyright information: The images in this commit were created by me
and are distributed under CC BY-SA 4.0 (this is conveyed in the image files)
with the exception of the "police officer icon at the center of the censored.png" 
which is from the Symbola font package http://users.teilar.gr/~g1951d/ which
carries the following licensing terms: "Fonts and documents in this site are not 
pieces of property or merchandise items; they carry no trademark, copyright, 
license or other market tags; they are free for any use."

rburchell commented:
I'd be interested in how it looks on other platforms too, but that generally looks pretty good. :)

rburchell commented on src/ui/qml/NetworkSetupWizard.qml:
```diff
@@ -93,43 +93,95 @@ ApplicationWindow {
+                Image {
+                    anchors.horizontalCenter: parent.horizontalCenter
+                    // Ideally we would use the svg here, but I cannot work out a way to get it to
+                    // display as nice as the png does.
```
what's the problem here?

(SVGs are kind of a good thing, since then you get retina support, say, for free)

special replied:
One useful hint is to use `sourceSize { width: n; height: n }` at the desired size for SVGs. That should make it render at the right resolution.

s-rah replied:
Thanks I will give that a try.

special commented on src/ui/qml/NetworkSetupWizard.qml:
```diff
@@ -93,43 +93,95 @@ ApplicationWindow {
+
+                    Image {
+                        anchors.horizontalCenter: parent.horizontalCenter
+                        source: "../icons/connection_ok.svg"
```
sourceSize may be useful on these as well

special commented on src/ui/qml/NetworkSetupWizard.qml:
```diff
@@ -93,43 +93,95 @@ ApplicationWindow {
+                    }
+
+                    Label {
+                        width: 200
```
Avoid hardcoding the width; it leads to subtle layout issues and scalability problems. In this case, you could either:

1. Use a RowLayout instead of Row, put `Layout.fillWidth: true` on the Columns
2. Set the Column's width to `(parent.width-1)/2`

In either case, you'd use `width: parent.width` for these items.

special commented:
Here's OS X/Retina:

![ricochet 2x](https://cloud.githubusercontent.com/assets/80188/8198601/414795c6-1466-11e5-95e4-8464b6cf2ce9.png)

The text wrapping makes it very crowded! I think this needs a lot more padding everywhere. I think even your screenshot could use more blank space. Don't hesitate to make the window larger and add lots of whitespace if it comes out looking cleaner.

special commented on src/ui/qml/NetworkSetupWizard.qml:
```diff
@@ -93,43 +93,95 @@ ApplicationWindow {
+
+                    Label {
+                        width: 200
+                        text: qsTr("This computer's Internet connection is free of obstacles.")
```
I've never really liked this phrase. What do you think of "connection is unfiltered"? "connection is free" sounds too much like it's talking about money, I suppose. Hmm.

s-rah replied:
I'd like to avoid complex terms here especially with negative prefixes. How about "connection is <b>not</b> censored, filtered and does not use a proxy?" to mimic the right hand side. Although I think a positive phrase would be better....

Although there aren't many positive options...clear?  "connection is clear of obstacles".

special replied:
Hmm, yes, positive is better. "not censored ..." is too similar to the other side.

"This computer's Internet connection is clear"
"This computer's Internet connection is clear and open"
"This computer's Internet connection is not blocked"
"This computer can access the Internet without restrictions"
"This computer can freely access the Internet"

s-rah replied:
> "This computer can freely access the Internet" 

Is probably my favourite there. Close second:"This computer can access the Internet without restrictions"

I'll have a play around and a think.

special commented on src/ui/qml/NetworkSetupWizard.qml:
```diff
@@ -93,43 +93,95 @@ ApplicationWindow {
+
+                    Label {
+                        width: 200
+                        text: qsTr("This computer's Internet connection is censored, filtered, or proxied.")
```
maybe "or uses a proxy" would be better than "or proxied"

s-rah commented:
Updated diff taking into account, comments, new screenshot in commit messages reflecting what it now looks like on Linux (Ubuntu) - would be interested to see how this looks elsewhere. Much more white space now (Maybe too much...I'm not sure...I've gone back and forth).

Also removed hardcoding width and switched to sourceSize which fixed the logo display issue.

rburchell commented on src/ui/qml/NetworkSetupWizard.qml:
```diff
@@ -93,43 +93,110 @@ ApplicationWindow {
+                Image {
+                    anchors.horizontalCenter: parent.horizontalCenter
+                    source: "../icons/ricochet.svg"
+                    sourceSize.width: parent.width / 10
```
It's not a good idea to use bindings for sourceSize. Every time it changes, the image will be reloaded (so every time parent.width changes..) which can cause a bad UX on resize (bad performance)

rburchell commented on src/ui/qml/NetworkSetupWizard.qml:
```diff
@@ -93,43 +93,110 @@ ApplicationWindow {
-                anchors.horizontalCenter: parent.horizontalCenter
-                text: qsTr("Configure")
-                onClicked: window.openConfig()
+                Column {
```
So this is just supposed to be a spacer?

    Item {
        width: parent.width / 5
        height: 1
    }

should perform the job OK (you just need to assign both a width and height to force it to take space).

rburchell commented:
Screenshot looks nice. I haven't tried it myself :)
